### PR TITLE
Set versioned to `True` on MetricsDataSet

### PR DESCRIPTION
**Pull request by limdauto**
## Checklist

- [ ] Read the [contributing](https://github.com/quantumblacklabs/kedro/blob/master/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/quantumblacklabs/kedro/blob/master/RELEASE.md) file
- [ ] Added tests to cover my changes

## Notice

- [ ] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

File: kedro/extras/datasets/tracking/metrics_dataset.py [hunk 1]
```diff
@@ -59,12 +59,14 @@ class MetricsDataSet(JSONDataSet):
 
     """
 
+    VERSIONED_FLAG_KEY = True
+
     # pylint: disable=too-many-arguments
     def __init__(
         self,
         filepath: str,
+        version: Version,
```
Comment (AntonyMilneQB) [inline kedro/extras/datasets/tracking/metrics_dataset.py:68]: Do we still need to overwrite `__init__` here? `JSONDataSet` has `version: Version = None` - not sure why this is now a compulsory argument here? 🤔

**Review by AntonyMilneQB — COMMENTED** [review]
Great idea, I like this approach 👍  Seems cleaner than needing to check the dataset type.

File: kedro/extras/datasets/tracking/metrics_dataset.py (hunk 1 above)
Comment (limdauto) [reply kedro/extras/datasets/tracking/metrics_dataset.py:68]: Yea actually I forgot to put it back it. I was trying something else. This can be optional.

**Comment by MerelTheisenQB** [comment]
I like this solution! Do you want to add this in my PR or do you want me to do it?

**Comment by limdauto** [comment]
@MerelTheisenQB do you mind taking over 🙏 I'm a bit tied up with the viz stuff and will be off for 2 weeks

**Pull request closed without merging**